jujugui/static/gui/src/app/components/header-search/header-search.js

```diff
@@ -47,6 +47,19 @@ YUI.add('header-search', function() {
     },
 
     /**
+      Update the state when the app state changes.
+
+      @method componentWillReceiveProps
+    */
+    componentWillReceiveProps: function() {
+      var component = this.props.getAppState(
```

Comment:
Why is this necessary? The empty method should be being called to unmount this if there is no component defined so it shouldn't have to close itself.